src/Host/Client/Impl/rtvs/R/grid.R

```diff
@@ -35,9 +23,19 @@ grid_data <- function(x, rows, cols, row_selector) {
   if (!missing(row_selector)) {
       x <- x[row_selector(x),, drop = FALSE]
   }
-  x <- x[rows, cols]
+  x <- x[rows, cols, drop = FALSE]
+
+  # Process and format values column by column, then flatten the resulting list of character vectors.
+  max_length <- 100 - 3
```

Comment:
Definitely. I kept the hardcoded value for now because wiring up the option would touch a lot more stuff.